### PR TITLE
refactor(): refactor InputGroup to TypeScript

### PR DESCRIPTION
**Pull request by David-Cohen35**
A little stuck here - line 48 and 53 have some type errors im unsure how to solve

### CODE REVIEW

**Comment by reywright** [comment]
@David-Cohen35 let's pair on this next week; I think I was able to solve it, but it'll be best to walk through the approach I think.

**Comment by ivanhdez28** [comment]
`flexSettings` in your props is `undefined | 'shrink' | 'grow'`, and you're passing that to `getFlexSettings` function, which takes a `string[]`

File: src/components/InputGroup/index.tsx [hunk 1]
```diff
@@ -31,7 +32,7 @@ export const getFlexSettings = (childCount, userSettings) => {
  * @param {Object} props react props
  * @returns {ReactElement}
  */
-const InputGroup = ({ children, flexSettings }) => {
+const InputGroup = ({ children, flexSettings }: InputGroupProps): React.ReactNode => {
```
Comment (reywright) [inline src/components/InputGroup/index.tsx:35]: This can probably just be `JSX.Element` as you know it always will return a single one of them?

**Comment by reywright** [comment]
@David-Cohen35 first TypeScript PR completed 🙌

**Pull request merged**